### PR TITLE
test(chaoscenter): add unit tests for Loader component

### PR DESCRIPTION
**Pull request by Shahazadi-Shaguftha-Syed**
## Proposed changes
Adds unit test coverage for the `Loader` component in `chaoscenter/web/src/components/Loader`, which previously had zero tests. Covers all three render branches: loading state (with default and custom message), no-data state, and default children render.

## Types of changes
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices applies)

## Checklist
- [x] I have read the CONTRIBUTING doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
None.

## Special notes for your reviewer:
This PR only adds test coverage — no changes to component logic. All 6 tests pass locally. Happy to add more edge cases if reviewers see gaps.

### CODE REVIEW

File: chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          }}
+        >
+          <div>Content</div>
+        </Loader>
+      </TestWrapper>
+    );
+
+    expect(screen.getByTestId('loader')).toBeInTheDocument();
+    expect(screen.queryByText('Content')).not.toBeInTheDocument();
+  });
```
[the remark this answers is not in the document]
Comment (Shahazadi-Shaguftha-Syed) [reply chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx:74]: Updated — now asserts the actual NoDataCard message text renders, not just that children are absent.

File: chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    render(
+      <TestWrapper>
+        <Loader loading={true} small testId="loader">
+          <div>Content</div>
+        </Loader>
+      </TestWrapper>
+    );
+
+    expect(screen.getByTestId('loader')).toBeInTheDocument();
+  });
```
[the remark this answers is not in the document]
Comment (Shahazadi-Shaguftha-Syed) [reply chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx]: Updated — renamed the test and added real assertions (spinner + message render correctly, children don't show) instead of just checking the container exists.

File: chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        <Loader loading={true} message="Fetching experiments..." testId="loader">
+          <div>Content</div>
+        </Loader>
+      </TestWrapper>
+    );
+
+    expect(screen.getByText('Fetching experiments...')).toBeInTheDocument();
+  });
+
+  test('renders children when loading is false and no noData condition', () => {
```
[the remark this answers is not in the document]
Comment (Shahazadi-Shaguftha-Syed) [reply chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx]: Good catch, renamed to "renders children when loading is false and noData is not provided".

**Comment by Shahazadi-Shaguftha-Syed** [comment]
@PriteshKiri  Addressed all 3 Copilot review comments — strengthened the NoDataCard and small-prop test assertions, and renamed a test for clarity. All 6 tests passing locally. Ready for another look whenever you get a chance!

File: chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import '@testing-library/jest-dom/extend-expect';
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { TestWrapper } from 'utils/testUtils';
+import { Loader } from '../Loader';
+
```
[the remark this answers is not in the document]
Comment (Shahazadi-Shaguftha-Syed) [reply chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx:6]: Added jest.mock for @harnessio/icons so icon name and size are queryable as data attributes. Now asserts icon name is 'steps-spinner' and size is '26' for small mode and '32' for default mode.

File: chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      <TestWrapper>
+        <Loader loading={true} testId="loader">
+          <div>Content</div>
+        </Loader>
+      </TestWrapper>
+    );
+
+    expect(screen.getByTestId('loader')).toBeInTheDocument();
+    expect(screen.getByText('loading')).toBeInTheDocument();
+    expect(screen.queryByText('Content')).not.toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (Shahazadi-Shaguftha-Syed) [reply chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx:26]: Updated — now asserts the expected icon name ('steps-spinner') and default size (32) using the mocked icon.

File: chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    render(
+      <TestWrapper>
+        <Loader loading={true} message="Fetching experiments..." testId="loader">
+          <div>Content</div>
+        </Loader>
+      </TestWrapper>
+    );
+
+    expect(screen.getByText('Fetching experiments...')).toBeInTheDocument();
+  });
```
[the remark this answers is not in the document]
Comment (Shahazadi-Shaguftha-Syed) [reply chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx:41]: Updated — now also asserts the default 'loading' text and 'Content' are absent when a custom message is provided.

File: chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+      <TestWrapper>
+        <Loader loading={true} small testId="loader">
+          <div>Content</div>
+        </Loader>
+      </TestWrapper>
+    );
+
+    expect(screen.getByTestId('loader')).toBeInTheDocument();
+    expect(screen.getByText('loading')).toBeInTheDocument();
+    expect(screen.queryByText('Content')).not.toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (Shahazadi-Shaguftha-Syed) [reply chaoscenter/web/src/components/Loader/__tests__/Loader.test.tsx:107]: Updated — renamed the test and added assertions for both the icon name and size (26 vs 32) to directly validate the small behavior.

**Comment by Shahazadi-Shaguftha-Syed** [comment]
@PriteshKiri  Squashed all commits into a single clean commit on #5567 — should be easier to review and merge now. Could you take a look when you get a chance? Thanks!

**Comment by PriteshKiri** [comment]
@Shahazadi-Shaguftha-Syed could you please check the CI failures?

**Comment by Shahazadi-Shaguftha-Syed** [comment]
Hi @PriteshKiri, checked the failures. They're all from the Trivy vulnerability scanner failing on pre-existing CVEs in golang.org/x/net (used by authentication-server, graphql-server, and subscriber), nothing related to my changes, which only touch the frontend Loader component's unit tests. web-unit-tests, frontend-checks, and docker-build-frontend are all green. This looks like it'd affect any PR touching those backend services right now. Let me know if you'd like me to take on a separate task for it, or if it needs a maintainer to bump the dependency

**Comment by PriteshKiri** [comment]
@Shahazadi-Shaguftha-Syed, the static code analysis is failing. Could you please check?

**Comment by PriteshKiri** [comment]
@Shahazadi-Shaguftha-Syed could you please fix the conflicts?

**Comment by Shahazadi-Shaguftha-Syed** [comment]
@PriteshKiri  ready for review

**Comment by PriteshKiri** [comment]
Hey @Shahazadi-Shaguftha-Syed 
Could you please look into the CI failures?

**Comment by PriteshKiri** [comment]
@Shahazadi-Shaguftha-Syed could you please fix the merge conflicts?

**Comment by PriteshKiri** [comment]
@Shahazadi-Shaguftha-Syed any updates?

**Comment by Shahazadi-Shaguftha-Syed** [comment]
@PriteshKiri  working upon it sorry for delay

**Comment by Shahazadi-Shaguftha-Syed** [comment]
@PriteshKiri  can you please review whenever u get chance
